### PR TITLE
Fix IpSubnetFilterRule with IPv6 Default Route does not accept all IPv6 addresses

### PR DESCRIPTION
**Pull request by skyguard1**
Motivation:

In this issue(https://github.com/netty/netty/issues/11349 ),IpSubnetFilterRule needs to support ipv6 reserved addresses, such as 8000::, but the current implementation does not support

Modification:

Added support for default rule

Result:

Fixes https://github.com/netty/netty/issues/11349

### CODE REVIEW

File: handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java [hunk 1]
```diff
@@ -23,60 +23,68 @@
 import io.netty.channel.embedded.EmbeddedChannel;
 import io.netty.util.internal.SocketUtils;
 import org.junit.Assert;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
```
Comment (normanmaurer) [inline handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java:27]: @skyguard1 please move the static imports in a separate "block"

Comment (skyguard1) [reply handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java]: I've done it

**Review by normanmaurer — APPROVED** [review]

File: handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java [hunk 2]
```diff
@@ -29,54 +29,63 @@
...
 
     @Test
     public void testIp4SubnetFilterRule() throws Exception {
         IpSubnetFilterRule rule = new IpSubnetFilterRule("192.168.56.1", 24, IpFilterRuleType.ACCEPT);
         for (int i = 0; i <= 255; i++) {
-            Assert.assertTrue(rule.matches(newSockAddress(String.format("192.168.56.%d", i))));
+            assertTrue(rule.matches(newSockAddress(String.format("192.168.56.%d", i))));
         }
-        Assert.assertFalse(rule.matches(newSockAddress("192.168.57.1")));
+        assertFalse(rule.matches(newSockAddress("192.168.57.1")));
```
Comment (trustin) [inline handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java:64]: Thanks a lot for cleaning this up!

**Review by trustin — APPROVED** [review]

File: handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java (hunk 2 above)
Comment (skyguard1) [reply handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java:64]: it is my pleasure

**Pull request merged**